File: continuum-js/continuum-idl-js/src/api/HasQualifiedName.ts
/**
 * Interface for objects that have a qualified name
 */
export interface HasQualifiedName {

    /**
     * This is the namespace of this such as "org.kinotic.continuum.idl.api.schema"
     */
    namespace?: string | null

    /**
     * This is the name of this such as "Person", "Animal"
     */
    name: string

    /**
     * The fully qualified name of this such as "org.kinotic.continuum.idl.api.schema.Person"
     * @return the fully qualified name of this
     */
    getQualifiedName(): string

}

File: continuum-idl/src/main/java/org/kinotic/continuum/idl/api/schema/HasQualifiedName.java
package org.kinotic.continuum.idl.api.schema;

import com.fasterxml.jackson.annotation.JsonIgnore;

/**
 * Created by Navíd Mitchell 🤪 on 4/20/24.
 */
public interface HasQualifiedName {

    /**
     * This is the namespace of this such as "org.kinotic.continuum.idl.api.schema"
     * @return the namespace of this
     */
    String getNamespace();

    /**
     * This is the name of this such as "Person", "Animal"
     * @return the name of this
     */
    String getName();


    /**
     * The fully qualified name of this such as "org.kinotic.continuum.idl.api.schema.Person"
     * @return the fully qualified name of this
     */
    @JsonIgnore
    default String getQualifiedName(){
        return getNamespace() + "." + getName();
    }

}
